test/unit/BMath.t.sol

```diff
@@ -0,0 +1,539 @@
+    uint256 _spotPrice = bMath.calcSpotPrice(balanceIn, weightIn, balanceOut, weightOut, 0);
+
+    assertEq(_spotPrice, 1.333333333333333333e18);
+  }
+
+  function test_CalcSpotPriceWhenSwapFeeIsNonZero() external virtual {
+    // it should return correct value
+    //     (bi/wi * wo/bo) * (1 / (1 - sf))
+    //     (20/1 * 2/30) * (1 / (1 - 0.1)) = 1.481481481...
+    uint256 _spotPrice = bMath.calcSpotPrice(balanceIn, weightIn, balanceOut, weightOut, swapFee);
+
+    assertEq(_spotPrice, 1.481481481481481481e18);
+  }
+
+  function test_CalcOutGivenInRevertWhen_TokenWeightOutIsZero() external {
+    uint256 _weightOut = 0;
+
+    // it should revert
+    //     division by zero
+    vm.expectRevert(BNum.BNum_DivZero.selector);
+
+    bMath.calcOutGivenIn(balanceIn, weightIn, balanceOut, _weightOut, amountIn, swapFee);
+  }
+
+  function test_CalcOutGivenInRevertWhen_SwapFeeGreaterThanBONE(uint256 _swapFee) external {
+    _swapFee = bound(_swapFee, BONE + 1, type(uint256).max);
+
+    // it should revert
+    //     division by zero
+    vm.expectRevert(BNum.BNum_SubUnderflow.selector);
+
+    bMath.calcOutGivenIn(balanceIn, weightIn, balanceOut, weightOut, amountIn, _swapFee);
+  }
+
+  function test_CalcOutGivenInWhenSwapFeeEqualsBONE() external virtual {
+    uint256 _swapFee = BONE;
+
+    // it should return zero
+    uint256 _amountOut = bMath.calcOutGivenIn(balanceIn, weightIn, balanceOut, weightOut, amountIn, _swapFee);
+
+    assertEq(_amountOut, 0);
+  }
+
+  function test_CalcOutGivenInRevertWhen_TokenAmountInTooBig(uint256 _amountIn) external {
+    _amountIn = bound(_amountIn, type(uint256).max / (BONE - swapFee) + 1, type(uint256).max);
+
+    // it should revert
+    //     ai * (1 - sf) > uint256 max
+    vm.expectRevert(BNum.BNum_MulOverflow.selector);
+
+    bMath.calcOutGivenIn(balanceIn, weightIn, balanceOut, weightOut, _amountIn, swapFee);
+  }
+
+  function test_CalcOutGivenInRevertWhen_TokenBalanceInAndAmountInAreZero() external {
+    uint256 _balanceIn = 0;
+    uint256 _amountIn = 0;
+
+    // it should revert
+    //     bi + (ai * (1 - swapFee)) = 0
+    vm.expectRevert(BNum.BNum_DivZero.selector);
+
+    bMath.calcOutGivenIn(_balanceIn, weightIn, balanceOut, weightOut, _amountIn, swapFee);
+  }
+
+  function test_CalcOutGivenInRevertWhen_TokenBalanceInIsZeroAndSwapFeeEqualsBONE() external {
+    uint256 _balanceIn = 0;
+    uint256 _swapFee = BONE;
+
+    // it should revert
+    //     bi + (ai * (1 - swapFee)) = 0
+    vm.expectRevert(BNum.BNum_DivZero.selector);
+
+    bMath.calcOutGivenIn(_balanceIn, weightIn, balanceOut, weightOut, amountIn, _swapFee);
+  }
+
+  function test_CalcOutGivenInWhenTokenWeightInIsZero() external virtual {
+    uint256 _weightIn = 0;
+
+    // it should return zero
+    uint256 _amountOut = bMath.calcOutGivenIn(balanceIn, _weightIn, balanceOut, weightOut, amountIn, swapFee);
+
+    assertEq(_amountOut, 0);
+  }
+
+  modifier whenTokenWeightsAreEqual() {
+    _;
```

Comment:
i think there're many ways of implementing it:
- single manual: just add one more 'valid state' constant `equalWeights = 4.2`, and use that one on the method calls of this branch (no use of `whenTokenWeightsAreEqual` modifier either)
- fuzzed manual: fuzz a valid equal weight each time (no use either)
- fuzzed state: add a state variable and use the modifier to set it, and then use the state variable (not the fuzzed var!) to call the method.. which makes it a bit harder to track what's going on
- fuzzed memory: add a struct, fuzz the struct, pass it to the modifier and modify it on the fly, which is kinda an overkill for this adding 2 `bound` lines

i think is ok that this is a result of applying the bulloak template but doesn't impose a drawback in DX tbh. yes it does the different modifiers for branches of the same name, i'm more unconvinced about that redundancy

Reply:
i don't feel guilty for not using this feature of the implementation, i think it'll make more sense when Pool is created and you'll perhaps need to update a state, but for these pure methods i think it may be overkill, idk

Reply:
okay, it's no big deal. 

do you think it's best to then pass `--skip-modifiers` to `bulloak check` when we configure it as a CI step so we can get rid of them?